benchmark/Streamly/Benchmark/FileSystem/Handle.hs

```diff
@@ -0,0 +1,1351 @@
+
+data Handles = Handles Handle Handle
+
+mkBench :: NFData b => String -> IORef Handles -> IO b -> Benchmark
+mkBench name href action =
+    bench name $ perRunEnv (do
+            Handles inH outH <- readIORef href
+
+            -- close old handles
+            hClose inH
+            hClose outH
+
+            -- reopen
+            inh <- openFile inFileBig ReadMode
+            outh <- openFile outfile WriteMode
+
+            -- update
+            writeIORef href $ Handles inh outh
+        )
+        (\_ -> action)
+
+-------------------------------------------------------------------------------
+-- read chunked using toChunks
+-------------------------------------------------------------------------------
+
+-- | Get the last byte from a file bytestream.
+{-# INLINE toChunksLast #-}
+toChunksLast :: Handle -> IO (Maybe Word8)
+toChunksLast inh = do
+    let s = IFH.toChunks inh
+    larr <- S.last s
+    return $ case larr of
+        Nothing -> Nothing
+        Just arr -> IA.readIndex arr (A.length arr - 1)
+
+#ifdef INSPECTION
+inspect $ hasNoTypeClasses 'toChunksLast
+inspect $ 'toChunksLast `hasNoType` ''Step
+#endif
+
+-- | Count the number of bytes in a file.
+{-# INLINE toChunksSumLengths #-}
+toChunksSumLengths :: Handle -> IO Int
+toChunksSumLengths inh =
+    let s = IFH.toChunks inh
+    in S.sum (S.map A.length s)
+
+#ifdef INSPECTION
+inspect $ hasNoTypeClasses 'toChunksSumLengths
+inspect $ 'toChunksSumLengths `hasNoType` ''Step
+#endif
+
+-- | Count the number of lines in a file.
+{-# INLINE toChunksSplitOnSuffix #-}
+toChunksSplitOnSuffix :: Handle -> IO Int
+toChunksSplitOnSuffix = S.length . AS.splitOnSuffix 10 . IFH.toChunks
+
+#ifdef INSPECTION
+inspect $ hasNoTypeClasses 'toChunksSplitOnSuffix
+inspect $ 'toChunksSplitOnSuffix `hasNoType` ''Step
+#endif
+
+-- XXX use a word splitting combinator instead of splitOn and test it.
+-- | Count the number of words in a file.
+{-# INLINE toChunksSplitOn #-}
+toChunksSplitOn :: Handle -> IO Int
+toChunksSplitOn = S.length . AS.splitOn 32 . IFH.toChunks
+
+#ifdef INSPECTION
+inspect $ hasNoTypeClasses 'toChunksSplitOn
+inspect $ 'toChunksSplitOn `hasNoType` ''Step
+#endif
+
+-- | Sum the bytes in a file.
+{-# INLINE toChunksCountBytes #-}
+toChunksCountBytes :: Handle -> IO Word8
+toChunksCountBytes inh = do
+    let foldlArr' f z = runIdentity . S.foldl' f z . IA.toStream
+    let s = IFH.toChunks inh
+    S.foldl' (\acc arr -> acc + foldlArr' (+) 0 arr) 0 s
+
+#ifdef INSPECTION
+inspect $ hasNoTypeClasses 'toChunksCountBytes
+inspect $ 'toChunksCountBytes `hasNoType` ''Step
+#endif
+
+{-# INLINE toChunksWithBufferOfDecodeUtf8ArraysLenient #-}
+toChunksWithBufferOfDecodeUtf8ArraysLenient :: Handle -> IO ()
+toChunksWithBufferOfDecodeUtf8ArraysLenient inh =
+   S.drain
+     $ IUS.decodeUtf8ArraysLenient
+     $ IFH.toChunksWithBufferOf (1024*1024) inh
+
+#ifdef INSPECTION
+inspect $ hasNoTypeClasses 'decodeUtf8Lenient
```

Comment:
`decodeUtf8Lenient` does not exist I here? Maybe you meant `toChunksWithBufferOfDecodeUtf8ArraysLenient`?

Reply:
This has been addressed in `bb6954f1a13553037e25862e3341f55e4e3ea0a2`.